engine/table/src/main/java/io/deephaven/engine/table/impl/PushdownResult.java

```diff
@@ -35,6 +41,23 @@ public class PushdownResult implements SafeCloseable {
      */
     public static final long DEFERRED_DATA_INDEX_COST = 50_000L;
 
+    private static final boolean IS_CI = "true".equalsIgnoreCase(System.getenv().get("CI"));
+
+    /**
+     * Forces additional safety checks in {@link #ofUnsafe(RowSet, RowSet, RowSet)}. Controlled via configuration
+     * property "PushdownResult.forceValidation". Not a user-documented feature. {@code false} by default, except when
+     * environment variable "CI" is {@code true}.
+     */
+    @VisibleForTesting
+    static final boolean FORCE_VALIDATION =
+            Configuration.getInstance().getBooleanWithDefault("PushdownResult.forceValidation", IS_CI);
+
+    /**
+     * The selection. Retaining selection here makes the pushdown result "self-contained" which helps with composability
+     * and potential safety checks when results are combined.
+     */
+    private final WritableRowSet selection;
```

Comment:
We can keep arguing about keeping `selection`, but its only purpose is to provide safety checks and will never be used in practice.  IMO, we shouldn't bother keeping it.  `PushdownResult` is a very simple special-purpose internal structure that is no more complicated than `Pair<WritableRowset,WritableRowSet>`.